sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -666,7 +673,47 @@ def _raw_predict(self, X):
                     predict = predictor.predict
                 raw_predictions[k, :] += predict(X)
 
-        return raw_predictions
+    def _staged_raw_predict(self, X):
+        """Compute raw predictions of ``X`` for each iteration.
+
+        This method allows monitoring (i.e. determine error on testing set)
+        after each stage.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The input samples.
+
+        Returns
+        -------
+        raw_predictions : generator of ndarray of shape (n_samples, k)
+            The raw predictions of the input samples. The order of the
+            classes corresponds to that in the attribute :term:`classes_`.
+            Regression and binary classification are special cases with
+            ``k == 1``, otherwise ``k==n_classes``.
+        """
+        X = check_array(X, dtype=[X_DTYPE, X_BINNED_DTYPE],
+                        force_all_finite=False)
+        check_is_fitted(self)
+        if X.shape[1] != self.n_features_:
+            raise ValueError(
+                'X has {} features but this estimator was trained with '
+                '{} features.'.format(X.shape[1], self.n_features_)
+            )
+        n_samples = X.shape[0]
+        raw_predictions = np.zeros(
+            shape=(self.n_trees_per_iteration_, n_samples),
+            dtype=self._baseline_prediction.dtype
+        )
+        raw_predictions += self._baseline_prediction
+        for stage in range(len(self._predictors)):
```

Comment:
```suggestion
        for iter in range(len(self._predictors)):
```

We prefer the use of `iteration` instead of `stage` in this part of the code (I understand this departs from the other gradient boosting estimators, but let's keep consistent with this file). `staged_predict` and `staged_raw_predict` are OK though, but for variable names `iteration` or similar are preferable to `stage`

Reply:
Would use "iteration" as variable name.